src/radical/pilot/pmgr/launching/base.py

```diff
@@ -278,8 +283,17 @@ def _kill_pilots(self, pids):
     #
     def work(self, pilots):
 
-        if not isinstance(pilots, list):
-            pilots = [pilots]
+        pilots = ru.as_list(pilots)
+
+        # weed out pilots for which we have already received a cancel request
+        to_start = list()
+        for pilot in pilots:
+            if pilot['uid'] in self._cancelled:
+                to_start.append(pilot)
+                self.advance(pilot, rps.CANCELED, publish=True, push=False)
+                continue
```

Comment:
both actions are mixed together (starting and cancelling)
```python
            if pilot['uid'] in self._cancelled:
                self.advance(pilot, rps.CANCELED, publish=True, push=False)
            else:
                to_start.append(pilot)
```

Reply:
btw tests for this PR failed due to this mistake (pilot never started, since it is not in `_cancelled`)

Reply:
What the heck was I thinking??  Nothing apparently... :-P